### PR TITLE
Implemented _reverse and _slice, expanded _log functionality.

### PR DESCRIPTION
**Pull request by thijsmie**

### CODE REVIEW

File: src/gdash/scripts/_log/_log.gml [hunk 1]
```diff
@@ -1,4 +1,16 @@
 /// @desc Convenience method for show_debug_message(). Automatically convetrs arguments to strings.
 /// @param {Mixed} Message The message or value to log
-show_debug_message(string(argument0));
 
+var toshow = "";
+for(var i = 0; i < argument_count; i++) {
+	if (i != 0)
+		toshow += ", ";
+	if (is_array(argument[i])) {
+		toshow += "[" + _join(argument[i],", ") + "]";
```
Comment (twisterghost) [inline src/gdash/scripts/_log/_log.gml:9]: Will this error if the array isn't strings?

File: src/gdash/scripts/_reverse/_reverse.gml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+/*
+@example
+var myArray = _arrayOf(1, 2, 3);
+var reverseArray = _reverse(myArray);
+_isEqual(_arrayOf(3,2,1), reverseArray)
+// => true
+*/
+
+var result;
+var n = _length(argument0)
```
Comment (twisterghost) [inline src/gdash/scripts/_reverse/_reverse.gml:13]: Minor codestyle, but missing a semicolon here

**Comment by twisterghost** [comment]
Looks good overall. Will pull it down and try for myself tonight.

File: src/gdash/scripts/_log/_log.gml (hunk 1 above)
Comment (thijsmie) [reply src/gdash/scripts/_log/_log.gml:9]: No, the _join script converts the array content to a string.

**Comment by twisterghost** [comment]
Works great. Will merge. One thought: do you think `_slice` should be able to work for `ds_list`s as well, or just arrays? So far I've been trying to allow that flexibility where it seems useful.

**Comment by twisterghost** [comment]
Release [2.2.0](https://github.com/twisterghost/gdash/releases/tag/2.2.0) has been posted with these changes.

**Pull request merged**